crates/ntx-builder/src/builder/coordinator.rs

```diff
@@ -0,0 +1,115 @@
+    /// actors.
+    actor_registry: HashMap<NetworkAccountPrefix, mpsc::UnboundedSender<MempoolEvent>>,
+    /// Join set for managing actor tasks. When an actor task completes, the actor's corresponding
+    /// data from the registry is removed.
+    actor_join_set: JoinSet<ActorShutdownReason>,
+    /// Semaphore for limiting the number of concurrent transactions across all network accounts.
+    semaphore: Arc<Semaphore>,
+}
+
+impl Coordinator {
+    /// Creates a new coordinator with the specified maximum number of inflight transactions.
+    pub fn new(max_inflight_transactions: usize) -> Self {
+        Self {
+            actor_registry: HashMap::new(),
+            actor_join_set: JoinSet::new(),
+            semaphore: Arc::new(Semaphore::new(max_inflight_transactions)),
+        }
+    }
+
+    /// Spawns a new actor to manage the state of the provided network account.
+    ///
+    /// If the account is not a network account, the function returns Ok(()) without spawning an
+    /// actor.
+    #[tracing::instrument(name = "ntx.builder.spawn_actor", skip(self, origin, config))]
+    pub async fn spawn_actor(&mut self, origin: AccountOrigin, config: &AccountActorConfig) {
+        let account_prefix = origin.prefix();
+
+        // If an actor already exists for this account prefix, something has gone wrong.
+        if self.actor_registry.contains_key(&account_prefix) {
+            tracing::error!("account actor already exists for prefix: {}", account_prefix);
+        }
+
+        // Construct the actor and add it to the registry for subsequent messaging.
+        let (event_tx, event_rx) = mpsc::unbounded_channel();
+        let actor = AccountActor::new(origin, config, event_rx);
+        self.actor_registry.insert(account_prefix, event_tx);
+
+        // Run the actor.
+        let semaphore = self.semaphore.clone();
+        self.actor_join_set.spawn(Box::pin(actor.run(semaphore)));
+
+        tracing::info!("created actor for account prefix: {}", account_prefix);
+    }
+
+    /// Broadcasts an event to all account actors.
+    pub fn broadcast_event(&self, event: &MempoolEvent) {
+        self.actor_registry.iter().for_each(|(account_prefix, event_tx)| {
+            Self::send(event_tx, event, *account_prefix);
+        });
+    }
+
+    /// Gets the next result from the actor join set and then handles it depending on the
+    /// reason the actor shutdown.
+    pub async fn next(&mut self) -> anyhow::Result<()> {
+        let actor_result = self.actor_join_set.join_next().await;
+        match actor_result {
+            Some(Ok(shutdown_reason)) => match shutdown_reason {
+                ActorShutdownReason::AccountReverted(account_prefix) => {
+                    tracing::info!("account reverted: {}", account_prefix);
+                    self.actor_registry.remove(&account_prefix);
+                    Ok(())
+                },
+                ActorShutdownReason::EventChannelClosed => {
```

Comment:
Do we want to cleanup the `self.actor_registry` in this case, or do we assume if we error here it's a fatal result _always_?